web-api-db-ctrl/src/elastic.rs

```diff
@@ -48,35 +51,73 @@ pub(crate) async fn setup_legacy_tenant(
     default_index: &str,
     tenant_id: &TenantId,
 ) -> Result<(), Error> {
+    if does_tenant_index_exist(elastic, default_index).await? {
+        create_index_alias(elastic, default_index, tenant_id).await?;
+    } else {
+        create_tenant_index(elastic, tenant_id).await?;
+    }
+
+    Ok(())
+}
+
+pub(crate) async fn migrate_tenant_index(
+    elastic: &Client,
+    tenant_id: &TenantId,
+) -> Result<(), Error> {
+    if !does_tenant_index_exist(elastic, tenant_id).await? {
+        error!(
+            {%tenant_id},
+            "index for tenant doesn't exist, creating a new index"
+        );
+        create_tenant_index(elastic, tenant_id).await?;
+    }
+    //FIXME code to check if the index is a super-set of the expected index
+    //FIXME code for allowing updates to the ES schema, at lest if
+    //      incremental applicable of the change is possible
+    Ok(())
+}
+
+async fn does_tenant_index_exist(
+    elastic: &Client,
+    tenant_id: impl AsRef<str>,
+) -> Result<bool, Error> {
     let response = elastic
-        .with_index(default_index)
+        .with_index(tenant_id)
         .request(Method::GET, [], [])
```

Comment:
:art: we could reduce this to a head request for less traffic
```suggestion
        .request(Method::HEAD, [], [])
```